### ISSUE TITLE
code=541, text='Unexpected Exception' System.Net.Sockets.SocketException (110): Connection timed out

### ISSUE THREAD

**Issue by sean19941103** [rabbitmq/rabbitmq-dotnet-client#923]
Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.Net.Sockets.SocketException (110): Connection timed out
   at RabbitMQ.Client.Impl.InboundFrame.ReadFrom(NetworkBinaryReader reader)
   at RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration()
   at RabbitMQ.Client.Framing.Impl.Connection.MainLoop()   at RabbitMQ.Client.Impl.SessionBase.Transmit(Command cmd)
   at RabbitMQ.Client.Impl.ModelBase.ModelRpc(MethodBase method, ContentHeaderBase header, Byte[] body)
   at RabbitMQ.Client.Framing.Impl.Model._Private_ExchangeDeclare(String exchange, String type, Boolean passive, Boolean durable, Boolean autoDelete, Boolean internal, Boolean nowait, IDictionary`2 arguments)
   at RabbitMQ.Client.Impl.AutorecoveringModel.ExchangeDeclare(String exchange, String type, Boolean durable, Boolean autoDelete, IDictionary`2 arguments)
   at God.Lottery.Console.GetNumber1040.Program.PulishToExchange(OfficialModel model) in D:\AspNet\Lottery\God.Lottery.Console.GetNumber\God.Lottery.Console.GetNumber1040\Program.cs:line 168 |God.Lottery.Console.GetNumber1040.Program|

what happend of  this error, i have a online rabbitmq server and i set RequestedConnectionTimeout=20000 and RequestedHeartbeat=60 and AutomaticRecoveryEnabled=true so why still Connection timed out? and as long as i restart the app, it will be back to normal

**Comment by michaelklishin** [comment]
Thank you for your time.

Team RabbitMQ uses GitHub issues for specific actionable items engineers can work on. GitHub issues are not used for questions, investigations, root cause analysis, discussions of potential issues, etc (as defined by this team).

We get at least a dozen of questions through various venues every single day, often light on details.
At that rate GitHub issues can very quickly turn into a something impossible to navigate and make sense of even for our team. Because GitHub is a tool our team uses heavily nearly every day, the signal/noise ratio of issues is something we care about a lot.

Please post this to [rabbitmq-users](https://groups.google.com/forum/#!forum/rabbitmq-users).

Thank you.

**Comment by michaelklishin** [comment]
`System.Net.Sockets.SocketException (110): Connection timed out` is the line you are looking for. A TCP socket connection timed out. Timeouts are a sad fact of life in distributed systems. Restarting the app makes it close all of its connections and try to reconnect again. 

[Troubleshooting Networking](https://www.rabbitmq.com/troubleshooting-networking.html) is of limited use here but can give you some ideas. Only network monitoring can provide relevant information.

You can use Toxiproxy and its alternatives to simulate a major network slowdown event and how connection recovery behaves (it pretty much should fail and be retried) in the 6.x releases of this client. If/when you have evidence of recovery not being complete, please share the steps we can use to reproduce.

**Comment by sean19941103** [comment]
> `System.Net.Sockets.SocketException (110): Connection timed out` is the line you are looking for. A TCP socket connection timed out. Timeouts are a sad fact of life in distributed systems. Restarting the app makes it close all of its connections and try to reconnect again.
> 
> [Troubleshooting Networking](https://www.rabbitmq.com/troubleshooting-networking.html) is of limited use here but can give you some ideas. Only network monitoring can provide relevant information.
> 
> You can use Toxiproxy and its alternatives to simulate a major network slowdown event and how connection recovery behaves (it pretty much should fail and be retried) in the 6.x releases of this client. If/when you have evidence of recovery not being complete, please share the steps we can use to reproduce.

thank you sir

**Comment by fishburner** [comment]
When I did a websearch for: "AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host" I got to this post here.
I just wanted to share my solution beacuse it took me 6 hours of troubleshooting:
My installation with rabbitMQ 3.11.6 or 3.11.16 together with Erlang 26.0 created this problem.
after using Erlang 25.2 my problem was solved.
